src/vector/index.js

```diff
@@ -382,6 +382,7 @@ async function loadLanguage() {
     }
     try {
         await languageHandler.setLanguage(langs);
+        document.getElementById("root").setAttribute("lang", languageHandler.getCurrentLanguage());
```

Comment:
This should propably be done in matrix-react-sdk inside the languageHandler.js. Otherwise it doesn't get changed if the user changes the language in the settings.

Reply:
that is true, and then even more evidently you can't rely on the outermost html element having an id since riot isn't the only consumer of matrix-react-sdk

Reply:
You actually want [document.documentElement](https://developer.mozilla.org/en-US/docs/Web/API/Document/documentElement).

Reply:
@MTRNord Are you sure? Doesn't the entire page get reloaded when you change language, and, thus, `loadLanguage()` would be called again, updating the attribute?

Reply:
@MTRNord I tested changing the language, and it did as @pafcu described: the app reloads and `loadLanguage()` is called again.  If it is more correct to go into `languageHandler.js`, where in that module should it go?

Reply:
@pafcu Are you sure that will _always_ be the element we want?

Reply:
@aidalgol When would it not be?

Reply:
@pafcu Or embed it at all.  I just wanted to double check, as I am still rather unfamiliar with the Riot codebase.

Reply:
The problem with the reload is there propably will be point of time where it isn't happening anymore. At that point this would break. Other than that at this position it looks just like a workaround and not like a fix from my point of view

Reply:
I would place it at the end of this function: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/languageHandler.js#L215

Reply:
@MTRNord I disagree (but it's not super important, and can be fixed later): A library (matrix-react-sdk) should not set the global state (lang attribute) of the application using the library.

Reply:
@pafcu well the current would propably need a refactor as soon as the translation change uses states instead a full reload

Reply:
@MTRNord And the "put it into the sdk" would need a refactor as soon as the split between the sdk and riot-web is made more clear.

Perhaps the correct approach is to fire a `languageChanged` event that the application can listen to. But considering how small the current proposed change is, I'd say put it in, and change it if and when someone implements something more robust.

Reply:
@pafcu what about putting this (in another PR) inside a promise returned from setLanguage()? That would propably make more sense to do

Reply:
Does that solve the issue of an app-developer only wanting to change the language of the matrix-specifc part of their page, rather than the entire page?

Reply:
@pafcu no but the issue that a reload might not happen in the future on language changes. (Thats what my point is)